### PR TITLE
Switch more explicitly to test if None; do not make truthy assumptions

### LINKED ISSUE #241: Make check for columns parameter more robust
The conditional `if not self.columns` in this line

https://github.com/exasol/pyexasol/blame/6a10f581ff4ea408563deda9a27ec5a647460d8b/pyexasol/http_transport.py#L115

has replaced a check `if "columns" not in self.params`.

This can cause some problems in cases where the columns parameter is not a list, but list-like. One particular instance happens when you try to import a dataframe into a table with a subset of the table's columns. The following used to work:

    import_from_pandas(df, table, schema, import_params={'columns' : df.columns})

but now it needs to be updated to

    import_from_pandas(df, table, schema, import_params={'columns' : list(df.columns)})

which might not be entirely expected. 

This could be made better by checking if self.columns is None, or alternatively to cast the columns to a list when the SqlQuery is instantiated.

### PR DESCRIPTION
**Pull request by ArBridgeman**
closes #241 

# Checklist

Note: If any of the items in the checklist are not relevant to your PR, leave the box unchecked.

## For any Pull Request

Is the following correct:
* [x] the title of the Pull Request?
* [x] the title of the corresponding issue?
* [x] there are no other open [Pull Requests](../../../../pulls) for the same update/change? 
* [x] that the issue which this Pull Request fixes ("Fixes...") is mentioned?

## When Changes Were Made

Did you:
* [x] update the changelog?
* [x] update the implementation?
* [x] check coverage and add tests: unit tests and, if relevant, integration tests?
* [ ] update the User Guide & other documentation?
* [x] resolve any failing CI criteria (incl. Sonar quality gate)?

## When Preparing a Release

Have you:
* [ ] thought about version number (major, minor, patch)?
* [ ] checked Exasol packages for updates and resolved open vulnerabilities, if easily possible?

### CODE REVIEW

File: pyexasol/http_transport.py [hunk 1]
```diff
@@ -30,16 +31,16 @@ class SqlQuery:
...
-    csv_cols: Optional[list[str]] = None
+    csv_cols: Optional[Iterable[str]] = None
     encoding: Optional[str] = None
     format: Optional[str] = None
     null: Optional[str] = None
     row_separator: Optional[str] = None
 
     def _build_csv_cols(self) -> str:
-        if not self.csv_cols:
+        if self.csv_cols is None:
```
Comment (MarleneKress79789) [inline pyexasol/http_transport.py:43]: What happens if the iterables where set, but are empty? i think this would return false. is that what you want?

**Review by MarleneKress79789 — APPROVED** [review]
